### PR TITLE
update AWS family policies for mid-2023

### PR DESCRIPTION
**Pull request by jonathan-3play**
## what

* Updated to all EC2 families as of 2023-06-23 that AWS documentation indicate are Nitro-based or support inherent encryption in transit.

## why

* Discovered that many useful EC2 families that should be permitted were in fact blocked.

## references

* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/data-protection.html#encryption-transit

### CODE REVIEW

File: catalog/ec2-policies.yaml [hunk 1]
```diff
@@ -74,273 +116,65 @@
...
-      - x2iedn.8xlarge
-      - x2iedn.metal
-      - x2iedn.xlarge
-      - x2iezn.12xlarge
-      - x2iezn.2xlarge
-      - x2iezn.4xlarge
-      - x2iezn.6xlarge
-      - x2iezn.8xlarge
-      - x2iezn.metal
+      # updated 2023-06-23
```
Comment (milldr) [inline catalog/ec2-policies.yaml:119]: can you please list the full names? It should be returned by the command

```shell
aws ec2 describe-instance-types \
--filters Name=network-info.encryption-in-transit-supported,Values=true \
--query "InstanceTypes[*].[InstanceType]" --output text | sort
```

File: catalog/ec2-policies.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #
 # This policy denies instance types that aren't based on the Nitro system as documented in the following document:
 # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances
+# List of Nitro-enabled instances updated 2023-06-23
```
Comment (milldr) [inline catalog/ec2-policies.yaml:4]:
```suggestion
# List of Nitro-enabled instances updated 2023-06-23
#
# aws ec2 describe-instance-types \
# --filters Name=hypervisor,Values=nitro \
# --query "InstanceTypes[*].[InstanceType]" --output text | sort
```

Comment (milldr) [reply catalog/ec2-policies.yaml:4]: The following should match the list returned by this command

**Comment by jonathan-3play** [comment]
Revised listings based on explicit queries. Note that the Nitro query drops all of the metal instances even though [the docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances) indicate "The following bare metal instances are built on the Nitro System:" and list out many the query misses. Shall I add the metal instances in separately, or do you want to trust the query output's exclusions?

File: catalog/ec2-policies.yaml (hunk 2 above)
Comment (jonathan-3play) [reply catalog/ec2-policies.yaml:7]: Done. Though see comment below about missing metal instances.

File: catalog/ec2-policies.yaml (hunk 1 above)
Comment (jonathan-3play) [reply catalog/ec2-policies.yaml:551]: Done.

**Comment by max-lobur** [comment]
Please bump yaml_config to `1.0.2` at `examples/complete/main.tf` - this will fix linting

**Pull request merged**